### PR TITLE
Added patch scripts to enable releasing fork to existing PoB installs

### PR DESCRIPTION
**Pull request by edmundsmith**
Hi,
I've been using/tweaking PoB for a while now, and I was getting frustrated at some of the slowdown in being kept up to date or reviewing pull reqs on the master repo combined with the inability for a non-technical user to use a fork repo. As Openarl appears to be the only person with the sources to SimpleGraphics engine, installer, etc., while PoB's core was open source, if he didn't have time to make a public release then nobody else could make an easy-to-use public release either.
I saw that a few people had made efforts to re-create SimpleGraphics to allow for users to install their forks without needing dev mode/git, but these had stagnated. I started looking into how to make my own release packages, and after finding the graphics re-impls and skipping over install wizards to auto-install forks into dev mode I looked into how the `manifest.xml` and update system works.
From this, I created a python script to allow devs to make compatible `manifest.xml` files for their forks so that PoB can auto-update from the new forks, and I added a small PowerShell script for end users to use to switch their update sources from the Openarl repo to an updater-compatible fork.
The python script will only need to be run by the dev, so a Python dependency there is ok, while I used PowerShell for the end-user manifest-patcher script as Powershell 2.0 and higher should already be installed on every Windows 7 and higher machine (this is also why I used WebClient objects instead of the newer Invoke-RestMethod/WebRequest commands not included in PowerShell 2.0, the Windows 7 default version).

To use, `updatemanifest.py` will read the existing `manifest.xml` and output `manifest-updated.xml`, which can then be checked over/version incremented before replacing the old manifest. To add a new source file to the update system, it will need a <File> entry in the manifest, then have `updatemanifest.py` run again.
Once the remote repo contains the appropriate updated manifest file, end users can then run `patch_fork.ps1` to replace their installed PoB's manifest with this fork's updated manifest file. The next time PoB checks for an update, it will update itself from the fork's repo instead, using the fork's manifest.xml to check for which files to download and replace.
Once this is done, PoB will auto-update from the fork's repo as per usual (keeping the `manifest.xml` updated is therefore important, hence having the `updatemanifest` script).

You will want to change the `$fork` variable in `patch_forks.ps1` to point to your repo.

## TL;DR

These scripts allow a dev to make their repo PoB-update compatible (`updatemanifest.py`) and allow an end user to switch their regular, installed PoB's update channel from Openarl to any compatible fork (just edit the `$fork` variable in patch_fork.ps1`) and then PoB-update Just Works

### CODE REVIEW

**Comment by LocalIdentity** [comment]
I have this in the repo at the moment, but the patch_forks.ps1 does not always work. When my friends have tried it, it will lock them into dev mode as it does not add the `platform="win32" branch="master"` after the end of the version number. I also had to add the ability for the updatemanifest script to work on .txt files so the changelog would receive an updated manifest number

**Comment by edmundsmith** [comment]
Ah, I think I've found the issue. My manifest.xml in the PR/my fork had a change to include the `branch="master"` attribute in the xml's version tag. 

Since the logic to detect dev mode in Launch.lua is

```lua 
        if localManXML and not self.versionBranch and not self.versionPlatform then
		-- Looks like a remote manifest, so we're probably running from a repository
		-- Enable dev mode to disable updates and set user path to be the script path
		self.devMode = true
	end
```

(https://github.com/Openarl/PathOfBuilding/blob/8a712d8ecaa7eb2a828b5984286595f5216cc63d/Launch.lua#L51)

the existence of either a branch *or* a platform tag will stop the dev mode from enabling, which is why it was working for my testing on my dev fork where I've added the tag, but it won't yet be working for your friends using this fork which doesn't have it yet.

## to fix

all you should need is to add in the `branch="master"` attribute to the manifest.xml\PoBVersion\Version tag, like how I have set up in my fork https://github.com/edmundsmith/PathOfBuilding/blob/78446dca1621742af588906f6ef679e476a94fd8/manifest.xml#L2
